stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -57,6 +57,13 @@ struct Eigen{T,V,S<:AbstractMatrix,U<:AbstractVector,R<:AbstractVector} <: Facto
     Eigen{T,V,S,U,R}(values::AbstractVector{V}, vectors::AbstractMatrix{T}, vectorsl::AbstractMatrix{T}, unitary::Bool, rconde::R, rcondv::R) where {T,V,S,U,R} =
         new(values, vectors, vectorsl, unitary, rconde, rcondv)
 end
+Eigen{T}(F::Eigen) where {T} = Eigen(
+           convert(AbstractVector{T}, F.values),
+           convert(AbstractMatrix{T}, F.vectors),
+           convert(AbstractMatrix{T}, F.vectorsl),
+           F.unitary::Bool,
+           convert(AbstractVector{T}, F.rconde),
+           convert(AbstractVector{T}, F.rcondv))
```

Comment:
As you can see from the constructor right below, the `rcond*` fields are allowed to have different fields than the `vectors[l]`. Also, some fields may be complex while others are real, and I think we would want to preserve that. That's why I now believe such a conversion-constructor is too generic.